Panic with empty name query

test by `dig . @127.0.0.1` (I don't know why but received odd query)

```
panic: runtime error: integer divide by zero

goroutine 56 [running]:
blocky/lists.stringCache.contains(0x0, 0xcd6500, 0x0, 0x3)
	/src/lists/list_cache.go:63 +0x1e5
blocky/lists.(*ListCache).Match(0xc0003ce300, 0xcd6500, 0x0, 0xc0003c4ba0, 0x1, 0x1, 0x2000100, 0x0, 0x0)
	/src/lists/list_cache.go:223 +0x10d
blocky/resolver.(*BlockingResolver).matches(0xc000372500, 0xc0003c4ba0, 0x1, 0x1, 0xce9748, 0xc0003ce300, 0xcd6500, 0x0, 0x0, 0x0, ...)
	/src/resolver/blocking_resolver.go:323 +0x92
blocky/resolver.(*BlockingResolver).handleBlacklist(0xc000372500, 0xc0003c4ba0, 0x1, 0x1, 0xc00030a1e0, 0xc0001ea5b0, 0x4b2267, 0xc000038140, 0xc000030c20)
	/src/resolver/blocking_resolver.go:204 +0x446
blocky/resolver.(*BlockingResolver).Resolve(0xc000372500, 0xc00030a1e0, 0xc0003f4f30, 0x1, 0x1)
	/src/resolver/blocking_resolver.go:227 +0x87d
blocky/resolver.(*ConditionalUpstreamResolver).Resolve(0xc0000d46e0, 0xc00030a1e0, 0xc0003f53f8, 0x1, 0x1)
	/src/resolver/conditionall_upstream_resolver.go:110 +0x474
blocky/resolver.(*CustomDNSResolver).Resolve(0xc0000d46c0, 0xc00030a1e0, 0x70, 0xc0001ea310, 0x7bcb790d6bc561f7)
	/src/resolver/custom_dns_resolver.go:131 +0x4d4
blocky/resolver.(*MetricsResolver).Resolve(0xc000080d20, 0xc00030a1e0, 0x337c5dfb003f5a50, 0x606fede9, 0xc0003f5970)
	/src/resolver/metrics_resolver.go:26 +0x58
blocky/resolver.(*StatsResolver).Resolve(0xc000170390, 0xc00030a1e0, 0x10e9c60, 0x6, 0xc0003f5ad8)
	/src/resolver/stats_resolver.go:54 +0x42
blocky/resolver.(*QueryLoggingResolver).Resolve(0xc00014f680, 0xc00030a1e0, 0xc0001ea2a0, 0xc, 0xc0003f5c70)
	/src/resolver/query_logging_resolver.go:117 +0x19f
blocky/resolver.(*ClientNamesResolver).Resolve(0xc000080910, 0xc00030a1e0, 0x10, 0x10, 0xc000162901)
	/src/resolver/client_names_resolver.go:71 +0x1d1
blocky/resolver.(*IPv6Checker).Resolve(0xc0000a5b60, 0xc00030a1e0, 0xc000210480, 0xc00030a1e0, 0x1)
	/src/resolver/ipv6_checker.go:19 +0x67
blocky/server.(*Server).OnRequest(0xc0003860f0, 0xcfa040, 0xc000078180, 0xc000210480)
	/src/server/server.go:279 +0x13a
github.com/miekg/dns.HandlerFunc.ServeDNS(0xc00013a480, 0xcfa040, 0xc000078180, 0xc000210480)
	/go/pkg/mod/github.com/miekg/dns@v1.1.41/server.go:37 +0x44
github.com/miekg/dns.(*ServeMux).ServeDNS(0xc0000d4500, 0xcfa040, 0xc000078180, 0xc000210480)
	/go/pkg/mod/github.com/miekg/dns@v1.1.41/serve_mux.go:103 +0x5d
github.com/miekg/dns.(*Server).serveDNS(0xc0000dd320, 0xc0002a0000, 0x11, 0xffff, 0xc000078180)
	/go/pkg/mod/github.com/miekg/dns@v1.1.41/server.go:651 +0x2fd
github.com/miekg/dns.(*Server).serveUDPPacket(0xc0000dd320, 0xc0003d2260, 0xc0002a0000, 0x11, 0xffff, 0xcf7cc8, 0xc00000e1c8, 0xc0000d4360, 0x0, 0x0)
	/go/pkg/mod/github.com/miekg/dns@v1.1.41/server.go:591 +0xed
created by github.com/miekg/dns.(*Server).serveUDP
	/go/pkg/mod/github.com/miekg/dns@v1.1.41/server.go:521 +0x395
```

--- a/lists/list_cache.go
+++ b/lists/list_cache.go
@@ -60,6 +60,10 @@ func (cache stringCache) elementCount() int {
 
 func (cache stringCache) contains(searchString string) bool {
 	searchLen := len(searchString)
+	if searchLen == 0 {
+		return false
+	}
+
 	searchBucketLen := len(cache[searchLen]) / searchLen
 	idx := sort.Search(searchBucketLen, func(i int) bool {
 		return cache[searchLen][i*searchLen:i*searchLen+searchLen] >= searchString

--- a/lists/list_cache_test.go
+++ b/lists/list_cache_test.go
@@ -43,6 +43,19 @@ var _ = Describe("ListCache", func() {
 	})
 
 	Describe("List cache and matching", func() {
+		When("Query with empty", func() {
+			It("should not panic", func() {
+				lists := map[string][]string{
+					"gr0": {emptyFile.Name()},
+				}
+				sut := NewListCache(BLACKLIST, lists, 0)
+
+				found, group := sut.Match("", []string{"gr0"})
+				Expect(found).Should(BeFalse())
+				Expect(group).Should(BeEmpty())
+			})
+		})
+
 		When("List is empty", func() {
 			It("should not match anything", func() {
 				lists := map[string][]string{

